src/inspector_agent.cc

```diff
@@ -477,6 +488,40 @@ void AgentImpl::InstallInspectorOnProcess() {
   v8::Local<v8::Object> inspector = v8::Object::New(env->isolate());
   READONLY_PROPERTY(process, "inspector", inspector);
   env->SetMethod(inspector, "wrapConsoleCall", InspectorWrapConsoleCall);
+  if (options_.wait_for_connect()) {
+    env->SetMethod(inspector, "callAndPauseOnStart", CallAndPauseOnStart);
+  }
+}
+
+// static
+void AgentImpl::CallAndPauseOnStart(
+      const v8::FunctionCallbackInfo<v8::Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  v8::HandleScope scope(env->isolate());
+  if (args.Length() < 2) {
+    return env->ThrowError("First argument for a "
+                           "inspector.callAndPauseOnStart call "
+                           "should be a function and the second one should be "
+                           "a thisArg");
+
+  } else if (!args[0]->IsFunction()) {
+    return env->ThrowError("First argument for a "
+                           "inspector.callAndPauseOnStart call "
+                           "should be a function");
+  }
```

Comment:
Just `CHECK(args.Length() > 1)` and `CHECK(args[0]->IsFunction())`?  This is only called by built-in code, right?